helix-core/src/main/java/org/apache/helix/task/WorkflowContext.java

```diff
@@ -81,15 +85,20 @@ public void setJobState(String job, TaskState s) {
     if (states == null) {
       states = new TreeMap<String, String>();
       _record.setMapField(WorkflowContextProperties.JOB_STATES.name(), states);
+      markWorkflowContextAsModified();
+    }
+    if (!s.name().equals(states.get(job))) {
```

Comment:
states.get(job) could be null, right? If the previous block happens.

Reply:
Yes, it could happen. But since (!noNull.equal(null)) will be true, then we will go to the if statement. Right?